### PR TITLE
API: Unified request parameter handling

### PR DESCRIPTION
**Pull request by annando**
We are now processing (nearly) all requests throufh a unified functionality. Now it doesn't matter anymore if an incoming API request had been done via PUT, PATCH or POST or if it had been done as formdata or JSON.

### CODE REVIEW

File: src/Module/Api/Mastodon/Lists.php [hunk 1]
```diff
@@ -78,13 +78,16 @@ public static function post(array $parameters = [])
 
 	public static function put(array $parameters = [])
 	{
-		$data = self::getPutData();
+		$request = self::getRequest([
+			'title'          => '', // The title of the list to be updated.
+			'replies_policy' => '', // Enumerable oneOf followed list none.
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/Lists.php]: "followed list none"?

File: src/Module/BaseApi.php [hunk 2]
```diff
@@ -142,25 +142,28 @@ public static function unsupported(string $method = 'all')
 	 * @return array request data
 	 */
 	public static function getRequest(array $defaults) {
```
Comment (MrPetovan) [inline src/Module/BaseApi.php]: Code standards: Please add a new line before method opening braces.
```suggestion
	public static function getRequest(array $defaults)
	{
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Api/Mastodon/Lists.php (hunk 1 above)
Comment (annando) [reply src/Module/Api/Mastodon/Lists.php]: It's a direct copy and paste.

Comment (MrPetovan) [reply src/Module/Api/Mastodon/Lists.php:83]: Without the original formatting it doesn't work as well.
```suggestion
			'replies_policy' => '', // Enumerable. One of: "followed", "list", or "none".
```

File: src/Module/Api/Mastodon/Statuses.php [hunk 3]
```diff
@@ -46,21 +46,22 @@ public static function post(array $parameters = [])
...
-		$scheduled_at   = $data['scheduled_at'] ?? ''; // Currently unsupported, but maybe in the future
-		$language       = $data['language'] ?? '';
+		$request = self::getRequest([
+			'status'         => '',    // Text content of the status. If media_ids is provided, this becomes optional. Attaching a poll is optional while status is provided.
+			'media_ids'      => [],    // Array of Attachment ids to be attached as media. If provided, status becomes optional, and poll cannot be used.
+			'poll'           => [],    // Poll data. If provided, media_ids cannot be used, and poll[expires_in] must be provided.
+			'in_reply_to_id' => 0,     // ID of the status being replied to, if status is a reply
+			'sensitive'      => false, // Mark status and attached media as sensitive?
+			'spoiler_text'   => '',    // Text to be shown as a warning or subject before the actual content. Statuses are generally collapsed behind this field.
+			'visibility'     => '',    // Visibility of the posted status. Enumerable oneOf public, unlisted, private, direct.
```
Comment (MrPetovan) [inline src/Module/Api/Mastodon/Statuses.php]:
```suggestion
			'visibility'     => '',    // Visibility of the posted status. One of: "public", "unlisted", "private" or "direct".
```

File: src/Module/Api/Mastodon/Lists.php (hunk 1 above)
Comment (MrPetovan) [reply src/Module/Api/Mastodon/Lists.php]: Without the original formatting it doesn't work as well.
```suggestion
			'replies_policy' => '', // One of: "followed", "list", or "none".
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**